From 645b4b83408027fcfa31e4a12eb09861c1a70a77 Mon Sep 17 00:00:00 2001
From: Steven Downer <grinch@outlook.com>
Date: Tue, 1 Oct 2013 22:48:23 -0500
Subject: [PATCH] (AEM) Update POM.xml

- Change 'api.version' from 'unknown' to 'v1_6_R3'
- Change 'defaultGoal' from 'package' to 'clean package'
- Change 'shadedPattern' to remove versioned classpaths
- Change mapper input and output to match a static name
- Change 'target' and 'source' from Java 1.6 to Java 1.7
- Add 'finalName' as 'Spigot' for jar name
- Add 'commons-lang' dependency
- Add relocation for CraftServer for plugin compatibility checks
- Update 'maven-compiler-plugin' from '2.3.2' to '3.1'
- Update 'maven-shade-plugin' from '1.4' to '2.1'

diff --git a/pom.xml b/pom.xml
index dc0a64b..c1b0f94 100644
--- a/pom.xml
+++ b/pom.xml
@@ -2,23 +2,15 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <parent>
-    <groupId>org.spigotmc</groupId>
-    <artifactId>spigot-parent</artifactId>
-    <version>dev-SNAPSHOT</version>
-    <relativePath>../pom.xml</relativePath>
-  </parent>
-
-  <groupId>org.spigotmc</groupId>
+  <groupId>net.aemservers</groupId>
   <artifactId>spigot</artifactId>
   <packaging>jar</packaging>
   <version>1.6.4-R2.1-SNAPSHOT</version>
   <name>Spigot</name>
-  <url>http://www.spigotmc.org</url>
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <api.version>unknown</api.version>
+    <api.version>v1_6_R3</api.version>
     <junit.version>4.11</junit.version>
     <minecraft.version>1.6.4</minecraft.version>
     <minecraft_version>1_6_R3</minecraft_version>
@@ -42,7 +34,7 @@
 
   <dependencies>
     <dependency>
-      <groupId>org.spigotmc</groupId>
+      <groupId>net.aemservers</groupId>
       <artifactId>spigot-api</artifactId>
       <version>${project.version}</version>
       <type>jar</type>
@@ -96,6 +88,11 @@
       <artifactId>gson</artifactId>
       <version>2.1</version>
     </dependency>
+    <dependency>
+      <groupId>org.ow2.asm</groupId>
+      <artifactId>asm-commons</artifactId>
+      <version>4.0</version>
+    </dependency>
     <!-- testing -->
     <dependency>
       <groupId>junit</groupId>
@@ -133,7 +130,8 @@
 
   <!-- This builds a completely 'ready to start' jar with all dependencies inside -->
   <build>
-    <defaultGoal>install</defaultGoal>
+    <defaultGoal>clean package</defaultGoal>
+    <finalName>${project.name}</finalName>
     <plugins>
       <plugin>
         <groupId>com.lukegb.mojo</groupId>
@@ -194,7 +192,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <version>2.0</version>
+        <version>2.1</version>
         <executions>
           <execution>
             <phase>package</phase>
@@ -205,7 +203,7 @@
               <relocations>
                 <relocation>
                   <pattern>org.bouncycastle</pattern>
-                  <shadedPattern>net.minecraft.v${minecraft_version}.org.bouncycastle</shadedPattern>
+                  <shadedPattern>net.minecraft.org.bouncycastle</shadedPattern>
                 </relocation>
                 <relocation>
                   <pattern>joptsimple</pattern>
@@ -227,6 +225,10 @@
                   <pattern>com.google.gson</pattern>
                   <shadedPattern>org.bukkit.craftbukkit.libs.com.google.gson</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>org.objectweb.asm</pattern>
+                  <shadedPattern>net.aemservers.internal.libs.org.objectweb.asm</shadedPattern>
+                </relocation>
                 <!-- These contain native libraries, and cannot be properly relocated
                 <relocation>
                   <pattern>org.fusesource</pattern>
@@ -240,17 +242,25 @@
                   <pattern>com.mysql</pattern>
                   <shadedPattern>org.bukkit.craftbukkit.libs.${project.dependencies[6].version}.com.mysql</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>org.bukkit.craftbukkit</pattern>
+                  <shadedPattern>org.bukkit.craftbukkit.v${minecraft_version}</shadedPattern>
+                  <excludes>
+                    <exclude>org.bukkit.craftbukkit.Main*</exclude>
+                  </excludes>
+                </relocation>
+                <relocation>
+                  <pattern>net.minecraft.server</pattern>
+                  <shadedPattern>net.minecraft.server.v${minecraft_version}</shadedPattern>
+                </relocation>
                 !-->
+                <!--  AEM CraftServer relocation -->
                 <relocation>
                     <pattern>org.bukkit.craftbukkit</pattern>
                     <shadedPattern>org.bukkit.craftbukkit.v${minecraft_version}</shadedPattern>
-                    <excludes>
-                        <exclude>org.bukkit.craftbukkit.Main*</exclude>
-                    </excludes>
-                </relocation>
-                <relocation>
-                    <pattern>net.minecraft.server</pattern>
-                    <shadedPattern>net.minecraft.server.v${minecraft_version}</shadedPattern>
+                    <includes>
+                        <include>org.bukkit.craftbukkit.CraftServer*</include>
+                    </includes>
                 </relocation>
               </relocations>
             </configuration>
@@ -260,10 +270,11 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>2.3.2</version>
+        <version>3.1</version>
         <configuration>
-          <source>1.6</source>
-          <target>1.6</target>
+          <source>1.7</source>
+          <target>1.7</target>
+          <encoding>${project.build.sourceEncoding}</encoding>
         </configuration>
       </plugin>
       <plugin>
@@ -315,8 +326,8 @@
             </goals>
             <configuration>
               <maps>${basedir}/target/maps.yml</maps>
-              <input>${basedir}/target/${project.artifactId}-${project.version}.jar</input>
-              <original>${basedir}/target/unmapped-${project.artifactId}-${project.version}.jar</original>
+              <input>${basedir}/target/${project.name}.jar</input>
+              <original>${basedir}/target/unmapped-${project.name}.jar</original>
             </configuration>
           </execution>
         </executions>
-- 
1.8.3.msysgit.0

